File: June/14-06-2024/Armstrong Numbers.java
//{ Driver Code Starts
// Initial Template for Java
import java.io.*;
import java.util.*;

class GFG {
    public static void main(String args[]) throws IOException {
        BufferedReader read = new BufferedReader(new InputStreamReader(System.in));
        int t = Integer.parseInt(read.readLine());
        while (t-- > 0) {
            int n = Integer.parseInt(read.readLine());
            Solution ob = new Solution();

            System.out.println(ob.armstrongNumber(n));
        }
    }
}
// } Driver Code Ends


// User function Template for Java
class Solution {
    static String armstrongNumber(int n) {
    //Your Code Start
         // Convert the number to a string to easily iterate through each digit
        String numStr = Integer.toString(n);
        
        // Number of digits
        int numDigits = numStr.length();
        
        // Compute the sum of each digit raised to the power of the number of digits
        int sumOfPowers = 0;
        for (char digitChar : numStr.toCharArray()) {
            int digit = Character.getNumericValue(digitChar);
            sumOfPowers += Math.pow(digit, numDigits);
        }
        
        // Check if the sum of powers is equal to the original number
        if (sumOfPowers == n) {
            return "true";
        } else {
            return "false";
        }

      //Your Code End
    }
}

File: June/14-06-2024/Armstrong Numbers.py
#User function Template for python3

class Solution:
    def armstrongNumber (self, n):
#Your Code Start
            # Convert the number to a string to easily iterate through each digit
        num_str = str(n)
    
    # Number of digits
        num_digits = len(num_str)
    
    # Compute the sum of each digit raised to the power of the number of digits
        sum_of_powers = sum(int(digit) ** num_digits for digit in num_str)
    
    # Check if the sum of powers is equal to the original number
        return "true" if sum_of_powers == n else "false"

#Your Code End



#{ 
 # Driver Code Starts
#Initial Template for Python 3
if __name__ == '__main__':
    t = int(input())
    for _ in range(t):
        n = input()
        n = int(n)
        ob = Solution()
        print(ob.armstrongNumber(n))

# } Driver Code Ends
